File: src/main/java/Piruru/cards/FlameBarrierButCold.java
package Piruru.cards;

import Piruru.abstracts.PiruruCard;
import Piruru.powers.FlameBarrierButColdP;
import com.megacrit.cardcrawl.characters.AbstractPlayer;
import com.megacrit.cardcrawl.core.CardCrawlGame;
import com.megacrit.cardcrawl.localization.CardStrings;
import com.megacrit.cardcrawl.monsters.AbstractMonster;

import static Piruru.Piruluk.makeID;

public class FlameBarrierButCold extends PiruruCard {
    private static final CardStrings cardStrings = CardCrawlGame.languagePack.getCardStrings(
            makeID(FlameBarrierButCold.class));
    private static final int COST = 2;
    private static final CardType TYPE = CardType.SKILL;
    private static final CardRarity RARITY = CardRarity.UNCOMMON;
    private static final CardTarget TARGET = CardTarget.SELF;
    private static final int BLOCK = 12;
    private static final int BLOCK_UP = 4;
    private static final int COLD = 1;

    public FlameBarrierButCold() {
        super(cardStrings, COST, TYPE, RARITY, TARGET, 0, BLOCK_UP, 0, COST);
        baseBlock = block = BLOCK;
        magicNumber = baseMagicNumber = COLD;
    }

    @Override
    public void use(AbstractPlayer p, AbstractMonster m) {
        block();
        act(power(p, p, new FlameBarrierButColdP(p, magicNumber), magicNumber));
    }
}

File: src/main/kotlin/Cirno/cards/FlameBarrierButCold.kt
package Cirno.cards

import Cirno.Cirno.Statics.makeID
import Cirno.abstracts.CirnoCard
import Cirno.powers.FlameBarrierButColdP
import com.megacrit.cardcrawl.cards.AbstractCard
import com.megacrit.cardcrawl.characters.AbstractPlayer
import com.megacrit.cardcrawl.core.CardCrawlGame
import com.megacrit.cardcrawl.monsters.AbstractMonster


class FlameBarrierButCold : CirnoCard(cardStrings, COST, TYPE, RARITY, TARGET, 0, BLOCK_UP, 0, COST) {
    init {
        block = BLOCK
        baseBlock = block
        baseMagicNumber = COLD
        magicNumber = baseMagicNumber
    }

    override fun use(p: AbstractPlayer, m: AbstractMonster?) {
        block()
        act(power(p, p, FlameBarrierButColdP(p, magicNumber), magicNumber))
    }

    companion object {
        private val cardStrings = CardCrawlGame.languagePack.getCardStrings(
                makeID(FlameBarrierButCold::class.java))
        private val COST = 2
        private val TYPE = AbstractCard.CardType.SKILL
        private val RARITY = AbstractCard.CardRarity.UNCOMMON
        private val TARGET = AbstractCard.CardTarget.SELF
        private val BLOCK = 12
        private val BLOCK_UP = 4
        private val COLD = 1
    }
}
